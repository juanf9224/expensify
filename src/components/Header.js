import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Go To Dashboard Page</NavLink><br />
    <NavLink to="/create" activeClassName="is-active">Go To Add Page</NavLink><br />
    <NavLink to="/edit" activeClassName="is-active">Go Edit Page</NavLink> <br />
    <button onClick={startLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => {
  return {
    startLogout: () => dispatch(startLogout())
  }
}

export default connect(undefined, mapDispatchToProps)(Header);