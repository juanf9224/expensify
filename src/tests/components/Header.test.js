import { shallow } from 'enzyme';
import React from 'react';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('h1').text()).toBe('Expensify');
  expect(wrapper).toMatchSnapshot();
});
