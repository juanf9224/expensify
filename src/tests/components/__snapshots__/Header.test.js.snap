// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Go To Dashboard Page
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Go To Add Page
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Go Edit Page
  </NavLink>
   
  <br />
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
